src/codegen_fn.cpp

```
@@ -52,17 +53,11 @@ EnumVariantRegistry CodeGen::buildEnumVariantRegistry() const {
-    static const std::pair<const char*, ResourceKind> table[] = {
-        {"TcpListener", RK_TcpListener}, {"TcpStream", RK_TcpStream},
-        {"TlsStream", RK_TlsStream}, {"HttpRequest", RK_HttpRequest},
-        {"HttpResponse", RK_HttpResponse}, {"HttpClientResponse", RK_HttpClientResponse},
-        {"JsonValue", RK_JsonValue},
-        {"Thread", RK_Thread}, {"Lock", RK_Lock}, {"RWLock", RK_RWLock},
-        {"Semaphore", RK_Semaphore}, {"Barrier", RK_Barrier},
-        {"AtomicInt", RK_AtomicInt}, {"AtomicBool", RK_AtomicBool},
-    };
-    for (auto &[name, rk] : table)
-        if (typeName == name) { resource_sets_[rk].insert(val); return; }
+    int rk = ResourceKindRegistry::instance().lookupByTypeName(typeName);
```

Clang (14.0.6)

[warning] variable 'rk' is not initialized

(cppcoreguidelines-init-variables)